pkg/steps/release/promote.go

```diff
@@ -214,16 +214,29 @@ func getPromotionPod(imageMirrorTarget map[string]string, namespace string, date
 	sort.Strings(keys)
 
 	var images []string
+	targets := sets.NewString()
 	for _, k := range keys {
 		if quayTags, err := tagsInQuay(imageMirrorTarget[k], k, date); err != nil {
 			logrus.WithField("key", k).WithError(err).
 				Warn("Failed to get the tag in quay.io and skipped the promotion to quay for this image")
 		} else {
 			for _, quayTag := range quayTags {
+				if targets.Has(quayTag) {
+					logrus.WithField("source", imageMirrorTarget[k]).WithField("target", quayTag).
+						Warn("Ignored a duplicated target to quay")
+					continue
+				}
 				images = append(images, fmt.Sprintf("%s=%s", imageMirrorTarget[k], quayTag))
+				targets.Insert(quayTag)
 			}
```

Comment:
Promotion targets are unique, so only the names based on hashes can ever conflict.  We don't need to check the others.

Reply:
Noticed this too. This problem will be gone if I do https://github.com/openshift/ci-tools/pull/3474 right.